### PR TITLE
Added cgroups v2 support to DockerHardeningScript

### PR DESCRIPTION
**Pull request by chrisbalmer**
- Added support for cgroups v2
- Fixed support for memory check_type=allocate
- Updated the Docker image to: demisto/python3:3.10.4.29342

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
When using Ubuntu 22.04 the memory check was failing due to Ubuntu changing to cgroups v2. The file path with the max RAM value was changed to a shared path. This update adds support for cgroups v2 without removing support for cgroups v1. 

While adding this support, I noticed the memory_check argument options did not match the code for allocate. I updated the code to use the `allocate` option instead of `allocation`. If a user supplied `allocate` like the argument list options supplied, the cgroup check was actually being used.

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 1]
```diff
@@ -37,7 +38,7 @@ def check_memory(target_mem: str, check_type: str) -> str:
         str -- error string if failed
     """
     size = mem_size_to_bytes(target_mem)
-    if check_type == "allocation":
+    if check_type == "allocate":
```
Comment (JasBeilin) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:41]: Can you please refer us to this change documentation?

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Thank you for your contribution!
 Please check my comment below so we can move forward :)

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py (hunk 1 above)
Comment (chrisbalmer) [reply Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:41]: I believe the original `allocation` was a typo as all of the documentation refers to `allocate`. Links to docs below:

[DockerHardeningCheck Script on XSOAR Dev Site](https://xsoar.pan.dev/docs/reference/scripts/docker-hardening-check#inputs)

> The memory check type to perform: cgroup - check memory cgroup configuration, allocate - try allocating actual memory and verify that the allocation fails. Note the allocate test on some configurations may cause the container to be killed by the linux memory manager and the whole test will then time out.

[Script YML Showing Argument Options](https://github.com/demisto/content/blob/a4b9233fbfda2e5243cdc1844fe9593b11386a22/Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.yml#L19)

>   predefined:
>  - cgroup
>  - allocate

[Comments for the function in code](https://github.com/demisto/content/blob/a4b9233fbfda2e5243cdc1844fe9593b11386a22/Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py#L34)

> check_type {str} -- How to check either: cgroup (check configuration of cgroup) or allocate (check actual allocation)

**Comment by JasBeilin** [comment]
Hi @chrisbalmer, please sync your master with demisto/Content master branch so the conflicts will be resolved and we can merge :)

**Pull request closed without merging**